addons/mail/static/src/core/common/thread_model.js

```diff
@@ -175,23 +109,62 @@ export class Thread extends Record {
         },
     });
     invitedMembers = Record.many("ChannelMember");
-    chatPartner = Record.one("Persona");
-    composer = Record.one("Composer", { inverse: "thread", onDelete: (r) => r.delete() });
+    chatPartner = Record.one("Persona", {
+        /** @this {import("models").Thread} */
+        compute() {
+            if (this.type !== "chat") {
+                return undefined;
+            }
+            for (const member of this.channelMembers) {
+                if (
+                    !member.persona?.eq(this._store.self) ||
+                    (this.channelMembers.length === 1 && member.persona?.eq(this._store.self))
+                ) {
+                    return member.persona;
+                }
+            }
+        },
+    });
+    composer = Record.one("Composer", {
+        compute: () => ({}),
+        inverse: "thread",
+        onDelete: (r) => r.delete(),
+    });
     counter = 0;
     /** @type {string} */
     custom_channel_name;
     /** @type {string} */
     description;
     followers = Record.many("Follower");
-    selfFollower = Record.one("Follower");
+    selfFollower = Record.one("Follower", {
+        /** @this {import("models").Thread} */
+        onAdd(r) {
+            r.followedThread = this;
+        },
+        onDelete: (r) => (r.followedThread = undefined),
```

Comment:
deleting the thread should delete the follower completely

Reply:
`onDelete()` refers to removing the `selfFollower` in relation, this hook is just there to unsync with a virtual inverse field.

Note that we don't use `inverse` because `selfFollower` and `followers` are inverse of `followedThread` but there's no support for more than 1 field per Model for `inverse`.